tests/integ/ml/test_transformer.py

```diff
@@ -0,0 +1,524 @@
+        res,
+        [
+            Row(
+                A=1,
+                B=4,
+                C=5,
+                ONEHOTARRAY="[\n  6,\n  [\n    0,\n    2,\n    4\n  ],\n  1\n]",
+            ),
+            Row(
+                A=2,
+                B=4,
+                C=5,
+                ONEHOTARRAY="[\n  6,\n  [\n    1,\n    2,\n    4\n  ],\n  1\n]",
+            ),
+            Row(
+                A=2,
+                B=3,
+                C=6,
+                ONEHOTARRAY="[\n  6,\n  [\n    1,\n    3,\n    5\n  ],\n  1\n]",
+            ),
+        ],
+    )
+
+
+def test_sparse_onehotencoder(session):
+    df = session.create_dataframe(
+        [[1, 4, 5], [2, 4, 5], [2, 3, 6]], schema=["a", "b", "c"]
+    )
+    onehotencoder = OneHotEncoder()
+    onehotencoder.input_cols = ["A", "B", "C"]
+    onehotencoder.output_cols = ["A_ONEHOT", "B_ONEHOT", "C_ONEHOT"]
+    model = onehotencoder.fit(df)
+
+    # test category table
+    df_table = session.table(model._category_table_name["A"])
+    Utils.check_answer(
+        df_table,
+        [Row(STATES_A=1, STATES_A_ENCODER=0), Row(STATES_A=2, STATES_A_ENCODER=1)],
+    )
+
+    res = model.transform(df, sparse=False)
+    Utils.check_answer(
+        res,
+        [
+            Row(
+                A=1,
+                B=4,
+                C=5,
+                A_ONEHOT="[\n  1,\n  0\n]",
+                B_ONEHOT="[\n  1,\n  0\n]",
+                C_ONEHOT="[\n  1,\n  0\n]",
+            ),
+            Row(
+                A=2,
+                B=4,
+                C=5,
+                A_ONEHOT="[\n  0,\n  1\n]",
+                B_ONEHOT="[\n  1,\n  0\n]",
+                C_ONEHOT="[\n  1,\n  0\n]",
+            ),
+            Row(
+                A=2,
+                B=3,
+                C=6,
+                A_ONEHOT="[\n  0,\n  1\n]",
+                B_ONEHOT="[\n  0,\n  1\n]",
+                C_ONEHOT="[\n  0,\n  1\n]",
+            ),
+        ],
+    )
+
+
+def test_sparse_merge_onehotencoder(session):
+    df = session.create_dataframe(
+        [[1, 4, 5], [2, 4, 5], [2, 3, 6]], schema=["a", "b", "c"]
+    )
+    onehotencoder = OneHotEncoder()
+    onehotencoder.input_cols = ["A", "B", "C"]
+    onehotencoder.output_cols = ["ONEHOTARRAY"]
+    model = onehotencoder.fit(df)
+    # test category table
+    df_table = session.table(model._category_table_name["A"])
+    Utils.check_answer(
+        df_table,
+        [Row(STATES_A=1, STATES_A_ENCODER=0), Row(STATES_A=2, STATES_A_ENCODER=1)],
+    )
+
+    res = model.transform(df, sparse=False)
+    Utils.check_answer(
+        res,
+        [
+            Row(A=1, B=4, C=5, ONEHOTARRAY="[\n  1,\n  0,\n  1,\n  0,\n  1,\n  0\n]"),
+            Row(A=2, B=4, C=5, ONEHOTARRAY="[\n  0,\n  1,\n  1,\n  0,\n  1,\n  0\n]"),
+            Row(A=2, B=3, C=6, ONEHOTARRAY="[\n  0,\n  1,\n  0,\n  1,\n  0,\n  1\n]"),
+        ],
+    )
+
+
+def test_input_cols_output_cols_not_match(session):
```

Comment:
Let's rename this test as `test_tranformer_input_cols_output_cols_negative` and have all negative cases here? Like no input columns, no output columns, etc.